bftengine/src/bftengine/FullNodeReplica.cpp

```diff
@@ -0,0 +1,295 @@
+      MsgCode::Checkpoint, std::bind(&FullNodeReplica::messageHandler<CheckpointMsg>, this, std::placeholders::_1));
+  msgHandlers_->registerMsgHandler(
+      MsgCode::ClientRequest,
+      std::bind(&FullNodeReplica::messageHandler<ClientRequestMsg>, this, std::placeholders::_1));
+  msgHandlers_->registerMsgHandler(
+      MsgCode::StateTransfer,
+      std::bind(&FullNodeReplica::messageHandler<StateTransferMsg>, this, std::placeholders::_1));
+  metrics_.Register();
+
+  SigManager::init(config_.replicaId,
+                   config_.replicaPrivateKey,
+                   config_.publicKeysOfReplicas,
+                   concord::crypto::KeyFormat::HexaDecimalStrippedFormat,
+                   ReplicaConfig::instance().getPublicKeysOfClients(),
+                   concord::crypto::KeyFormat::PemFormat,
+                   {{repsInfo->getIdOfOperator(),
+                     ReplicaConfig::instance().getOperatorPublicKey(),
+                     concord::crypto::KeyFormat::PemFormat}},
+                   *repsInfo);
+
+  // Register status handler for Full Node replica
+  registerStatusHandlers();
+  bft::communication::StateControl::instance().setGetPeerPubKeyMethod(
+      [&](uint32_t id) { return SigManager::instance()->getPublicKeyOfVerifier(id); });
+}
+
+void FullNodeReplica::start() {
+  ReplicaForStateTransfer::start();
+  size_t sendAskForCheckpointMsgPeriodSec = config_.get("concord.bft.ro.sendAskForCheckpointMsgPeriodSec", 30);
+  askForCheckpointMsgTimer_ = timers_.add(
+      std::chrono::seconds(sendAskForCheckpointMsgPeriodSec), Timers::Timer::RECURRING, [this](Timers::Handle) {
+        if (!this->isCollectingState()) {
+          sendAskForCheckpointMsg();
+        }
+      });
+  msgsCommunicator_->startMsgsProcessing(config_.replicaId);
+}
+
+void FullNodeReplica::stop() {
+  timers_.cancel(askForCheckpointMsgTimer_);
+  ReplicaForStateTransfer::stop();
+}
+
+void FullNodeReplica::onTransferringCompleteImp(uint64_t newStateCheckpoint) {
+  lastExecutedSeqNum = newStateCheckpoint * checkpointWindowSize;
+
+  fn_metrics_.last_executed_seq_num_.Get().Set(lastExecutedSeqNum);
+  last_executed_seq_num_ = lastExecutedSeqNum;
+}
+
+void FullNodeReplica::onReportAboutInvalidMessage(MessageBase *msg, const char *reason) {
+  fn_metrics_.received_invalid_msg_++;
+  LOG_WARN(GL,
+           "Node " << config_.replicaId << " received invalid message from Node " << msg->senderId()
+                   << " type=" << msg->type() << " reason: " << reason);
+}
+void FullNodeReplica::sendAskForCheckpointMsg() {
+  fn_metrics_.sent_ask_for_checkpoint_msg_++;
+  LOG_INFO(GL, "sending AskForCheckpointMsg");
+  auto msg = std::make_unique<AskForCheckpointMsg>(config_.replicaId);
+  for (auto id : repsInfo->idsOfPeerReplicas()) send(msg.get(), id);
+}
+
+template <>
+void FullNodeReplica::onMessage<StateTransferMsg>(std::unique_ptr<StateTransferMsg> msg) {
+  ReplicaForStateTransfer::onMessage(move(msg));
+}
+
+template <>
```

Comment:
add a comment and TODO - this function is temporary, inherited from RO replica.
Should be removed later.
(not reviewing)

Reply:
will add a comment  in the beginning of the file and in the PR description i will mention about the same